vignettes/tweak-page-hooks.Rmd

```diff
@@ -0,0 +1,105 @@
+---
+title: "Custom pkgdown tweak page hooks"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Custom pkgdown tweak page hooks}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+This article will go over the following:
+
+* The process for using custom hooks in R packages
+* Example how to create custom hooks
+
+For more information about hooks, please read `help("getHook", "base")`.
+
+## Custom hooks
+
+Currently, the pkgdown do not allow set up any custom action, the changes apply during the executing will modify the `tweak_page` function to execute extra code base on the hook content.
```

Comment:
```suggestion
Currently,  `{pkgdown}` does not allow the set up of any custom action, the changes apply during the executing will modify the `tweak_page` function to execute extra code based on the hook content.
```

Reply:
I tried to wordsmith this a bit, but I'm not familiar with what we are doing here honestly.